source/slang/slang-parser.cpp

```diff
@@ -4706,9 +4706,45 @@ namespace Slang
         }
     }
 
+    static IRIntegerValue _foldIntegerPrefixOp(TokenType tokenType, IRIntegerValue value)
+    {
+        switch (tokenType)
+        {
+            case TokenType::OpInc:      return ++value;
+            case TokenType::OpDec:      return --value;
```

Comment:
I may be missing something, but wouldn't the `++` and `--` operators always be invalid on something that is a literal constant (since it can't also be an l-value), and if it *were* valid wouldn't trying to "fold" them like this actually drop the side effect?

Reply:
Actually yeah cos they'd have to be writable.... will remove.